crates/bevy_input/Cargo.toml

```diff
@@ -9,36 +9,72 @@ license = "MIT OR Apache-2.0"
 keywords = ["bevy"]
 
 [features]
-default = ["bevy_reflect"]
+default = ["std", "bevy_reflect", "bevy_ecs/async_executor", "smol_str"]
+
+# Functionality
+
+## Adds runtime reflection support using `bevy_reflect`.
 bevy_reflect = [
   "dep:bevy_reflect",
   "bevy_app/bevy_reflect",
   "bevy_ecs/bevy_reflect",
   "bevy_math/bevy_reflect",
 ]
-serialize = ["serde", "smol_str/serde"]
+
+## Adds serialization support through `serde`.
+serialize = [
+  "serde",
+  "smol_str/serde",
+  "bevy_ecs/serialize",
+  "bevy_math/serialize",
+]
+
+## Uses the small-string optimization provided by `smol_str`.
+smol_str = ["dep:smol_str", "bevy_reflect/smol_str"]
+
+# Platform Compatibility
+
+## Allows access to the `std` crate. Enabling this feature will prevent compilation
+## on `no_std` targets, but provides access to certain additional features on
+## supported platforms.
+std = [
+  "bevy_app/std",
+  "bevy_ecs/std",
+  "bevy_math/std",
+  "bevy_utils/std",
+  "bevy_reflect/std",
+]
+
+## `critical-section` provides the building blocks for synchronization primitives
+## on all platforms, including `no_std`.
+critical-section = ["bevy_app/critical-section", "bevy_ecs/critical-section"]
+
+## `portable-atomic` provides additional platform support for atomic types and
+## operations, even on targets without native support. 
+portable-atomic = ["bevy_app/portable-atomic", "bevy_ecs/portable-atomic"]
+
+## Uses the `libm` maths library instead of the one provided in `std` and `core`.
+libm = ["bevy_math/libm"]
 
 [dependencies]
 # bevy
 bevy_app = { path = "../bevy_app", version = "0.15.0-dev", default-features = false }
-bevy_ecs = { path = "../bevy_ecs", version = "0.15.0-dev", default-features = false, features = [
-  "serialize",
-] }
-bevy_math = { path = "../bevy_math", version = "0.15.0-dev", default-features = false, features = [
-  "rand",
```

Comment:
Seems that the `bevy_math/rand` feature just disappeared during these changes.
Is that intentional? (I mean I don't see a reason for bevy_input to need random numbers.)

Reply:
Yeah I deliberately left that feature off since I couldn't see anything in `bevy_input` that actually needed it.